internal/services/config/config_test.go

```diff
@@ -0,0 +1,266 @@
+  objectStorage:
+    type: posix
+    path: /agola/configstore/ost
+  web:
+    listenAddress: ":4002"
+
+runservice:
+  dataDir: /data/agola/runservice
+  etcd:
+    endpoints: "http://localhost:2379"
+  objectStorage:
+    type: posix
+    path: /agola/runservice/ost
+  web:
+    listenAddress: ":4000"
+
+gitserver:
+  dataDir: /data/agola/gitserver
+  gatewayURL: "http://localhost:8000"
+  web:
+    listenAddress: ":4003"`,
+		},
+		{
+			name: "test gateway, scheduler and notification",
+			services: []string{"gateway", "scheduler", "notification"},
+			in: `
+gateway:
+  apiExposedURL: "http://localhost:8000"
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  gitserverURL: "http://localhost:4003"
+
+  web:
+    listenAddress: ":8000"
+  tokenSigning:
+    method: hmac
+    key: supersecretsigningkey
+  adminToken: "admintoken"
+
+scheduler:
+  runserviceURL: "http://localhost:4000"
+
+notification:
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  etcd:
+    endpoints: "http://localhost:2379"
+
+configstore:
+  dataDir: 
+
+runservice:
+  dataDir: 
+
+gitserver:
+  dataDir:`,
+		},
+		{
+			name: "test gateway, scheduler and notification and gitserver without dataDir",
+			services: []string{"gateway", "scheduler", "notification", "gitserver"},
+			in: `
+gateway:
+  apiExposedURL: "http://localhost:8000"
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  gitserverURL: "http://localhost:4003"
+
+  web:
+    listenAddress: ":8000"
+  tokenSigning:
+    method: hmac
+    key: supersecretsigningkey
+  adminToken: "admintoken"
+
+scheduler:
+  runserviceURL: "http://localhost:4000"
+
+notification:
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  etcd:
+    endpoints: "http://localhost:2379"
+
+configstore:
+  dataDir: 
+
+runservice:
+  dataDir: 
+
+gitserver:
+  dataDir:`,
+  			err: errors.Errorf("git server dataDir is empty"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// create tmp dir
+			dir, err := ioutil.TempDir("", "ParseConfig")
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			defer os.RemoveAll(dir)
+			// write conf file
+			content := []byte(tt.in)
+			err = ioutil.WriteFile(path.Join(dir, "config.yml"), content, 0644)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			// read and test conf
+			if _, err := Parse(path.Join(dir, "config.yml"), tt.services); err != nil {
+				if tt.err == nil {
+					t.Fatalf("got error: %v, expected no error", err)
+				}
+				if errs, ok := err.(*util.Errors); ok {
+					if !errs.Equal(tt.err) {
+						t.Fatalf("got error: %v, want error: %v", err, tt.err)
+					}
+				} else {
```

Comment:
Probably this was copied by another test but this blocks isn't needed. services/config doesn't use util.Errors.